### PR TITLE
Add test for spread parameters after optional chaining

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Ref: https://github.com/babel/babel/issues/13001

I couldn't manually verify the Firefox and Safari data, but @jridgewell wrote that they work in https://github.com/babel/babel/issues/13001 so I copied the same versions for the other optional chaining features.

### CODE REVIEW

File: data-es2016plus.js [hunk 1]
```diff
@@ -4591,6 +4591,26 @@ exports.tests = [
...
+
+          return fn?.(...[], 1) === void undefined && fn?.(...[], ...[]) === void undefined && o.method?.(...[], 1) === void undefined && n?.method(...[], 1) === void undefined;
+        */},
+        res : {
+          ie11: false,
+          firefox10: false,
+          firefox52: false,
+          firefox73: false,
+          firefox74: true,
+          chrome89: false,
```
Comment (ljharb) [inline data-es2016plus.js:4611]: is there a version of chrome where this ever worked?

**Review by ljharb — APPROVED** [review]

File: data-es2016plus.js (hunk 1 above)
Comment (jridgewell) [reply data-es2016plus.js:4611]: I downloaded an old Chrome 80 build, and it failed there, too. I don't think it's ever been supported.

Comment (nicolo-ribaudo) [reply data-es2016plus.js:4611]: I'll add this so that older versions are marked as "no" and not as "unknown"

**Review by chicoxyzzy — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
Rebased to fix the merge conflict

**Pull request merged**